File: frontend/src/types/family.ts
import { Member } from './member';

export interface Family {
  id: string;
  community_id: string;
  name: string;
  description?: string;
  head_of_family?: string;
  created_at: string;
  updated_at: string;
}

export interface FamilyMember {
  id: string;
  family_id: string;
  member_id: string;
  role: string;
  created_at: string;
  updated_at: string;
}

export enum FamilyRoles {
  SPOUSE = 'Cônjuge',
  CHILD = 'Filho(a)',
  SIBLING = 'Irmão(ã)',
  PARENT = 'Pai/Mãe',
  GRANDPARENT = 'Avô/Avó',
  GRANDCHILD = 'Neto(a)',
  UNCLE_AUNT = 'Tio(a)',
  NEPHEW_NIECE = 'Sobrinho(a)',
  COUSIN = 'Primo(a)',
  OTHER = 'Outro'
}

export const FamilyRoleValues = {
  SPOUSE: 'spouse',
  CHILD: 'child',
  SIBLING: 'sibling',
  PARENT: 'parent',
  GRANDPARENT: 'grandparent',
  GRANDCHILD: 'grandchild',
  UNCLE_AUNT: 'uncle_aunt',
  NEPHEW_NIECE: 'nephew_niece',
  COUSIN: 'cousin',
  OTHER: 'other'
} as const; 

File: backend/internal/domain/family.go
package domain

import (
	"time"
)

type Family struct {
	ID           string    `json:"id" gorm:"primaryKey;type:uuid" db:"id"`
	CommunityID  string    `json:"community_id" gorm:"type:uuid;not null" db:"community_id"`
	Name         string    `json:"name" gorm:"not null" db:"name"`                      // Nome da família (ex: "Família Silva")
	Description  string    `json:"description" gorm:"type:text" db:"description"`       // Descrição ou notas sobre a família
	HeadOfFamily string    `json:"head_of_family" gorm:"type:uuid" db:"head_of_family"` // ID do membro que é o chefe da família
	CreatedAt    time.Time `json:"created_at" gorm:"not null" db:"created_at"`
	UpdatedAt    time.Time `json:"updated_at" gorm:"not null" db:"updated_at"`
}

type FamilyMember struct {
	ID        string    `json:"id" gorm:"primaryKey;type:uuid" db:"id"`
	FamilyID  string    `json:"family_id" gorm:"type:uuid;not null" db:"family_id"`
	MemberID  string    `json:"member_id" gorm:"type:uuid;not null" db:"member_id"`
	Role      string    `json:"role" gorm:"not null" db:"role"` // esposo, esposa, filho(a), irmão(ã), etc
	CreatedAt time.Time `json:"created_at" gorm:"not null" db:"created_at"`
	UpdatedAt time.Time `json:"updated_at" gorm:"not null" db:"updated_at"`
}

// Roles possíveis para membros da família
const (
	FamilyRoleSpouse      = "spouse"       // Cônjuge
	FamilyRoleChild       = "child"        // Filho(a)
	FamilyRoleSibling     = "sibling"      // Irmão(ã)
	FamilyRoleParent      = "parent"       // Pai/Mãe
	FamilyRoleGrandparent = "grandparent"  // Avô/Avó
	FamilyRoleGrandchild  = "grandchild"   // Neto(a)
	FamilyRoleUncleAunt   = "uncle_aunt"   // Tio(a)
	FamilyRoleNephewNiece = "nephew_niece" // Sobrinho(a)
	FamilyRoleCousin      = "cousin"       // Primo(a)
	FamilyRoleOther       = "other"        // Outro
)
